ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryLimitFunctionalTest.java

```diff
@@ -0,0 +1,491 @@
+                clusterFormation.router.getHost(), HighAvailabilityTestUtil.lagsReported(3), USER_CREDS);
+
+
+        final String sqlStreamPull = "SELECT * FROM " + USERS_STREAM + ";";
+        final String sqlLimit = "SELECT * FROM " + USERS_STREAM + " LIMIT " + numLimitRows + " ;";
+
+        //scan the entire stream first
+        final List<StreamedRow> rows_0 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlStreamPull, null, USER_CREDS);
+
+        //check that we got back all the rows
+        assertThat(rows_0, hasSize(HEADER + TOTAL_RECORDS));
+
+        //issue pull query on stream with limit
+        final List<StreamedRow> rows_1 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlLimit, null, USER_CREDS);
+
+        // check that we only got limit number of rows
+        assertThat(rows_1, hasSize(limitSubsetSize));
+
+        // Partition off the active
+        clusterFormation.active.getShutoffs().shutOffAll();
+
+        waitForRemoteServerToChangeStatus(
+                clusterFormation.router.getApp(),
+                clusterFormation.standBy.getHost(),
+                HighAvailabilityTestUtil::remoteServerIsUp,
+                USER_CREDS);
+        waitForRemoteServerToChangeStatus(
+                clusterFormation.router.getApp(),
+                clusterFormation.active.getHost(),
+                HighAvailabilityTestUtil::remoteServerIsDown,
+                USER_CREDS);
+
+
+        // issue stream scan after partitioning active
+        final List<StreamedRow> rows_2 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlStreamPull, null, USER_CREDS);
+
+        // check that we get all rows back
+        assertThat(rows_2, hasSize(HEADER + TOTAL_RECORDS));
+        assertThat(rows_0, is(matchersRowsAnyOrder(rows_2)));
+
+        // issue pull query with limit after partitioning active
+        final List<StreamedRow> rows_3 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlLimit, null, USER_CREDS);
+
+        // check that we only got limit number of rows
+        assertThat(rows_3, hasSize(limitSubsetSize));
+    }
+
+    private ClusterFormation findClusterFormation(
+            TestApp testApp0, TestApp testApp1, TestApp testApp2) {
+        ClusterFormation clusterFormation = new ClusterFormation();
+        ClusterStatusResponse clusterStatusResponse
+                = HighAvailabilityTestUtil.sendClusterStatusRequest(testApp0.getApp(), USER_CREDS);
+        ActiveStandbyEntity entity0 = clusterStatusResponse.getClusterStatus().get(testApp0.getHost())
+                .getActiveStandbyPerQuery().get(queryId);
+        ActiveStandbyEntity entity1 = clusterStatusResponse.getClusterStatus().get(testApp1.getHost())
+                .getActiveStandbyPerQuery().get(queryId);
+
+        // find active
+        if (!entity0.getActiveStores().isEmpty() && !entity0.getActivePartitions().isEmpty()) {
```

Comment:
getActivePartitions() is a set of topic partitions.  This only works with 1 partition.